File: php/testcode/headerinj_002.php
<?php
require_once __DIR__ . '/shared.php';

// vuln-code-snippet start php_headerinj_sanitized
function headerinj002(BenchmarkRequest $req): BenchmarkResponse {
    $input = $req->param('url');
    $sanitized = str_replace(["\r", "\n"], '', $input); // vuln-code-snippet safe-line php_headerinj_sanitized
    header("Location: " . $sanitized);
    return BenchmarkResponse::ok('Redirecting...');
}
// vuln-code-snippet end php_headerinj_sanitized

File: ruby/testcode/headerinj_002.rb
require_relative 'shared'

# vuln-code-snippet start ruby_headerinj_strip_crlf
def redirect_to_url_sanitized(req)
  input = req.param('url')
  safe_value = input.gsub(/[\r\n]/, '')  # vuln-code-snippet safe-line ruby_headerinj_strip_crlf
  BenchmarkResponse.new(302, '', { 'Location' => safe_value })
end
# vuln-code-snippet end ruby_headerinj_strip_crlf
